diff --git a/modules/product/commerce_product.module b/modules/product/commerce_product.module
index ba096c2..03b64af 100644
--- a/modules/product/commerce_product.module
+++ b/modules/product/commerce_product.module
@@ -117,7 +117,7 @@ function template_preprocess_commerce_product(array &$variables) {
   $product = $variables['elements']['#commerce_product'];
 
   $variables['product_entity'] = $product;
-  $variables['product_url'] = $product->toUrl();
+  $variables['product_url'] = ''; //$product->toUrl();
   $variables['product'] = [];
   foreach (Element::children($variables['elements']) as $key) {
     $variables['product'][$key] = $variables['elements'][$key];
@@ -140,7 +140,7 @@ function template_preprocess_commerce_product_variation(array &$variables) {
   $product = $product_variation->getProduct();
 
   $variables['product_variation_entity'] = $product_variation;
-  $variables['product_url'] = $product->toUrl();
+  $variables['product_url'] = ''; //$product->toUrl();
   $variables['product_variation'] = [];
   foreach (Element::children($variables['elements']) as $key) {
     $variables['product_variation'][$key] = $variables['elements'][$key];
diff --git a/modules/product/src/Entity/ProductVariation.php b/modules/product/src/Entity/ProductVariation.php
index 7f9b237..08a813d 100644
--- a/modules/product/src/Entity/ProductVariation.php
+++ b/modules/product/src/Entity/ProductVariation.php
@@ -352,6 +352,11 @@ class ProductVariation extends ContentEntityBase implements ProductVariationInte
       // Title generation is not possible before the parent product is known.
       return '';
     }
+    if (!is_object($this->getProduct())) {
+      if ($this->get('title')->value) {
+        return $this->get('title')->value;
+      }
+    }
 
     $product_title = $this->getProduct()->getTitle();
     if ($attribute_values = $this->getAttributeValues()) {

